### PR TITLE
Return an { message, path } object from custom types

### PR DESCRIPTION
**Pull request by davidchase**
ref #102

### CODE REVIEW

**Comment by panva** [comment]
Maybe a case for `string` can still be kept so that this is not a breaking change? (I know while below ^1.0.0 anything minor can still be breaking, but still, not too much effort to support all three cases (string, boolean and object).

**Comment by davidchase** [comment]
@panva thats a good point, we should keep string as a backwards compatibility option... I didn't even realize superstruct was not v1

File: package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "build:max": "rollup --config ./config/rollup-umd.js",
     "build:min": "rollup --config ./config/rollup-umd-min.js",
     "clean": "rm -rf ./lib ./umd ./node_modules",
-    "lint": "eslint '{src,test}/*' && prettier --list-different '**/*.{js,json,md}'",
+    "lint": "eslint \"{src,test}/*\" && prettier --list-different \"**/*.{js,json}\"",
```
Comment (davidchase) [inline package.json]: i have windows work machine so i need to wrap the eslint commands in double quotes to get it to work. if this is a problem i will remove

File: .eslintrc [hunk 2]
```diff
@@ -37,7 +37,7 @@
     "import/no-named-as-default": "error",
     "import/no-named-as-default-member": "error",
     "import/no-unresolved": "error",
-    "linebreak-style": "error",
+    "linebreak-style": ["error", "windows"],
```
Comment (davidchase) [inline .eslintrc]: same as below.. windows made me do it 😞 lol

File: src/kinds.js [hunk 3]
```diff
@@ -232,25 +232,36 @@ function func(schema, defaults, options) {
   const type = '<function>'
   const validate = (value = resolveDefaults(defaults), data) => {
     const result = schema(value, data)
```
Comment (davidchase) [inline src/kinds.js]: the logic was getting hairy keeping track of all of the supported types so i refactored a little bit let me know if it works

**Comment by davidchase** [comment]
ahh travis took a dive because of the eslint `line-breaks` differ on windows and linux/mac osx

**Comment by ianstormtaylor** [comment]
Thanks @davidchase!

**Pull request merged**